beacon-chain/p2p/discovery.go

```diff
@@ -205,19 +223,48 @@ func (s *Service) listenForNewNodes() {
 			// for new peers.
 			log.Trace("Not looking for peers, at peer limit")
 			time.Sleep(pollingPeriod)
+			if searchInProgress {
+				log.Debug("Searching for new active peers - success")
+			}
+			searchInProgress = false
 			continue
 		}
-		wantedCount := s.wantedPeerDials()
-		if wantedCount == 0 {
+
+		// Compute the number of new peers we want to dial.
+		activePeerCount, missingPeerCount := peersSummary(s.cfg.MaxPeers)
+
+		fields := logrus.Fields{
+			"currentPeerCount": activePeerCount,
+			"targetPeerCount":  s.cfg.MaxPeers,
+		}
+
+		if missingPeerCount == 0 {
 			log.Trace("Not looking for peers, at peer limit")
+			if searchInProgress {
+				log.WithFields(fields).Debug("Searching for new active peers - success")
+			}
 			time.Sleep(pollingPeriod)
+			searchInProgress = false
 			continue
 		}
+
+		if searchInProgress {
```

Comment:
why the `searchInProgress` boolean ? it seems like its only used for logging. Do we really need to differentiate it ?

Reply:
My specifications are:
- If we are starting / continuing / finishing a search for active peers, I want to know it in debug 
- If we are starting / continuing / finishing a search for peers for a given subnets, I want to know it in debug.

Reply:
Just to clarify, `listenForNewNodes` isn't used for searching peers on our subnets. It is a background discovery routine that we use for general peer discovery. If you want to log for a subnet search it needs to be in `FindPeersInSubnet` 

> If we are starting / continuing / finishing a search for active peers, I want to know it in debug

Is it sufficient to just know when you are still 'searching' for new peers ? You don't need the boolean for that. I don't see the value in tracking `starting` because it only happens once.

Reply:
> Just to clarify, `listenForNewNodes` isn't used for searching peers on our subnets. It is a background discovery routine that we use for general peer discovery. If you want to log for a subnet search it needs to be in `FindPeersInSubnet`

Yes, there is different logging for that. For a dedicated subnet we have:
```
DEBUG p2p: Searching for new peers for a subnet - success currentPeerCount=1 targetPeerCount=1 topic=/eth2/edf8b306/data_column_sidecar_101/ssz_snappy
```

> Is it sufficient to just know when you are still 'searching' for new peers ? You don't need the boolean for that. I don't see the value in tracking starting because it only happens once.

It only happens once (in the BN lifecycle) in the best case. If, during the life of the BN, the peer counts goes bellow the threshold, then `start` happens again.

==> Fixed in fa4f6933fdb548f4de2c37313a1462ab403f0097